osu.Game/Screens/Play/HUD/PerformancePointsCounter.cs

```diff
@@ -0,0 +1,173 @@
+            }
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            if (scoreProcessor != null)
+                scoreProcessor.NewJudgement += onNewJudgement;
+        }
+
+        private void onNewJudgement(JudgementResult judgement)
+        {
+            if (gameplayState?.Score == null || timedAttributes == null || timedAttributes.Length == 0)
+                return;
+
+            Debug.Assert(gameplayRuleset != null);
+
+            var attribIndex = Array.BinarySearch(timedAttributes, 0, timedAttributes.Length, new TimedDifficultyAttributes(judgement.HitObject.GetEndTime(), null));
+            if (attribIndex < 0)
+                attribIndex = ~attribIndex - 1;
+            attribIndex = Math.Clamp(attribIndex, 0, timedAttributes.Length - 1);
+
+            var ppProcessor = gameplayRuleset.CreatePerformanceCalculator(timedAttributes[attribIndex].Attributes, gameplayState.Score.ScoreInfo);
+            Current.Value = (int)Math.Round(ppProcessor?.Calculate() ?? 0, MidpointRounding.AwayFromZero);
+        }
+
+        protected override LocalisableString FormatCount(int count) => count.ToString(@"D");
+
+        protected override IHasText CreateText() => new TextComponent();
+
+        protected override void Dispose(bool isDisposing)
+        {
+            base.Dispose(isDisposing);
+
+            if (scoreProcessor != null)
+                scoreProcessor.NewJudgement -= onNewJudgement;
+
+            loadCancellationSource?.Cancel();
+        }
+
+        private class TextComponent : CompositeDrawable, IHasText
+        {
+            public LocalisableString Text
+            {
+                get => text.Text;
+                set => text.Text = value;
+            }
+
+            private readonly OsuSpriteText text;
+
+            public TextComponent()
+            {
+                AutoSizeAxes = Axes.Both;
+
+                InternalChild = new FillFlowContainer
+                {
+                    AutoSizeAxes = Axes.Both,
+                    Spacing = new Vector2(2),
+                    Children = new Drawable[]
+                    {
+                        text = new OsuSpriteText
+                        {
+                            Anchor = Anchor.BottomLeft,
+                            Origin = Anchor.BottomLeft,
+                            Font = OsuFont.Numeric.With(size: 16)
+                        },
+                        new OsuSpriteText
+                        {
+                            Anchor = Anchor.BottomLeft,
+                            Origin = Anchor.BottomLeft,
+                            Text = @"pp",
+                            Font = OsuFont.Numeric.With(size: 8)
+                        }
+                    }
+                };
+            }
+        }
+
+        // Todo: This class shouldn't exist, but requires breaking changes to allow DifficultyCalculator to receive an IBeatmap.
```

Comment:
I think at very least an issue should be created tracking this (or just pen in the follow up change sometime in the near future). Don't feel comfortable with it remaining in "todo never" inline todo land, especially since I'm actively trying to tighten the beatmap data flow..